searx/network/__init__.py

```diff
@@ -1,266 +1,137 @@
 # SPDX-License-Identifier: AGPL-3.0-or-later
 # lint: pylint
-# pylint: disable=missing-module-docstring, global-statement
-
-import asyncio
+# pylint: disable=global-statement
+"""Provide HTTP clients following the user configuration
+"""
 import threading
```

Comment:
Describe the API: the implementation uses httpx, but the API is mostly compatible with requests.

**mostly** because, may be there are some missing parts.
TODO : check https://www.python-httpx.org/compatibility/